### PR TITLE
More RP startup refactoring

### PR DESCRIPTION
**Pull request by jim-minter**
This set of commits removes autodiscovery of database, key vaults and DNS as these have occasionally given us startup problems in production.

In addition they begin to separate out use of the service key vault, making it easier for additional services *not* to use it.

There are a few cleanups as well.

### CODE REVIEW

**Comment by jim-minter** [comment]
have (hopefully) updated CI secrets to correspond with this PR

File: cmd/aro/monitor.go [hunk 1]
```diff
@@ -59,7 +61,20 @@ func monitor(ctx context.Context, log *logrus.Entry) error {
 		return err
 	}
 
-	key, err := _env.GetBase64Secret(ctx, env.EncryptionSecretName)
+	rpKVAuthorizer, err := _env.NewRPAuthorizer(_env.Environment().ResourceIdentifiers.KeyVault)
+	if err != nil {
+		return err
+	}
+
+	// TODO: should not be using the service keyvault here
```
Comment (mjudeikis) [inline cmd/aro/monitor.go:69]: Why? How monitor would get kubeconfig to access cluster? Or you expect us to maintain separate table with different kubeconfig for monitor?

**Review by mjudeikis — APPROVED** [review]
look ok. it is simplier than it looks

File: cmd/aro/monitor.go (hunk 1 above)
Comment (jim-minter) [reply cmd/aro/monitor.go:69]: There's some initial detail in the permission separation architecture doc.  The monitor should run as a separate identity and there will be a database token service in yet another identity.  Either everyone who needs access to the encryption secret will access it using their identity from a new shared key vault, or they will get it via REST API from the database token service.  The details depend on how we want to build towards encryption secret rotation.

**Pull request merged**